lib/background_worker/base.rb

```diff
@@ -72,109 +65,58 @@ def perform_in_background(method_name, options={})
         # Enqueue to the background queue
         BackgroundWorker.enqueue(self, method_name, options)
 
-        return options[:uid]
+        options[:uid]
       end
 
-
-      def rescue_reconnect
-        Resque.redis.client.disconnect
-        Resque.redis.client.reconnect
-        Rails.cache.reconnect
-      end
-
-
       # This method is called by the job runner
       #
       # It will just call your preferred method in the worker.
-      def perform(method_name, options={})
-        rescue_reconnect if defined?(Resque)
-        ActiveRecord::Base.verify_active_connections!
+      def perform(method_name, options = {})
+        fail ArgumentError, ":uid is required: Options given were: #{options.inspect}" if options['uid'].blank?
```

Comment:
Ok for the line being too long, but for the rest, Hound, have a chat to Rubocop and find some agreement, will you? Thanks!